modular_skyrat/master_files/code/modules/pain_shock/status_indicator.dm

```diff
@@ -0,0 +1,198 @@
+	. = ..()
+	var/atom/movable/screen/plane_master/runechat/status/status = locate() in client?.screen
+	if(!status)
+		return
+	if(value)
+		status.alpha = 255
+	else
+		status.alpha = 0
+/mob
+	var/status_enabled = TRUE
+/mob/living
+	var/list/status_indicators = null // Will become a list as needed.
+
+/* /mob/living/carbon/proc/is_critical()
+	if(HAS_TRAIT(src, TRAIT_CRITICAL_CONDITION))
+		return TRUE */
+
+/mob/living/proc/is_weakened()
+	if(HAS_TRAIT_FROM(src, TRAIT_FLOORED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || has_status_effect(/datum/status_effect/incapacitating/knockdown))
+		return TRUE
+
+/mob/living/proc/is_stunned()
+	if(HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) || HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, CHOKEHOLD_TRAIT))
+		return TRUE
+
+/mob/living/proc/is_paralyzed()
+	if(HAS_TRAIT_FROM(src, TRAIT_FLOORED, CHOKEHOLD_TRAIT) || HAS_TRAIT(src, TRAIT_CRITICAL_CONDITION) || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, STAMINA) || HAS_TRAIT_FROM(src, TRAIT_INCAPACITATED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) && HAS_TRAIT_FROM(src, TRAIT_IMMOBILIZED, TRAIT_STATUS_EFFECT(STAT_TRAIT)) && HAS_TRAIT_FROM(src, TRAIT_FLOORED, TRAIT_STATUS_EFFECT(STAT_TRAIT)))
+		return TRUE
+
+/mob/living/proc/is_unconcious()
+	if(HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+		return TRUE
+
+/mob/living/proc/is_confused()
+	if(has_status_effect(/datum/status_effect/confusion))
+		return TRUE
+
+/* /mob/living/carbon/proc/is_blind_status()
+	if(HAS_TRAIT_FROM(src, TRAIT_BLIND, STATUS_EFFECT_TRAIT) || HAS_TRAIT_FROM(src, TRAIT_BLIND, EYES_COVERED) && !HAS_TRAIT_FROM(src, TRAIT_BLIND, QUIRK_TRAIT)) // We don't want permanently blind users (those who took the trait) to have this forever.
+		return TRUE */ // Disabled, uncomment if we want this.
+
+/mob/living/carbon/death(gibbed) // On death, we clear the indiciators
+	..() // Call the TG death. Do not . = ..()!
+	for(var/iteration in status_indicators) // When we die, clear the indicators.
+		remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
+/* /datum/status_effect/incapacitating/on_apply()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 100 MILLISECONDS)
+/datum/status_effect/incapacitating/on_remove()
+	. = ..()
+	addtimer(CALLBACK(owner, /mob/living/.proc/status_sanity), 1000 MILLISECONDS) */
+
+/* /mob/living/carbon/apply_effect()
+	. = ..()
+	status_sanity() */
+
+/mob/living/carbon/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_CARBON_HEALTH_UPDATE, .proc/status_sanity)
+
+/* /mob/living/updatehealth()
+	. = ..()
+	status_sanity() */
+/mob/living/proc/status_sanity()
+	SIGNAL_HANDLER
+	if(stat == DEAD)
+		for(var/iteration in status_indicators) // When we die, clear the indicators.
+			remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
```

Comment:
This also probably could be its own proc, like, the for loop to remove them all, it should be something like `remove_all_status_indicators()`.

You also need to add a `return` there, otherwise they will, in fact, be re-applied regardless afterwards.
```suggestion
	if(stat == DEAD)
		for(var/iteration in status_indicators) // When we die, clear the indicators.
			remove_status_indicator(icon_state) // The indicators are named after their icon_state and type
		return
```

Reply:
well true, yeah. Death I am sure removes all status effects anyways but done next commit